docs/api/openapi.yaml

```diff
@@ -0,0 +1,651 @@
+# Copyright 2020 The Tekton Authors
```

Comment:
How did you generate this document?

Reply:
I used an API designer and added spec, https://www.apibldr.com and then did the test using https://editor-next.swagger.io/

Reply:
For the most part, it is manually written. Couldn't find a better way. There is no proper way to generate OpenAPI 3.0 spec from proto files.

Reply:
Would https://pkg.go.dev/github.com/zchee/protoc-gen-openapi work?

Reply:
> Would https://pkg.go.dev/github.com/zchee/protoc-gen-openapi work?

I tried this, did not work completely. Generated partial specs. Also since the project is from Istio, its add title as Istio APIs.

Reply:
This tool from the google/gnostic repo is the closest I found, but requires the Proto file to have proper descriptive comments.
https://github.com/google/gnostic/tree/main/cmd/protoc-gen-openapi

Reply:
As Sayan mentioned, zchee/protoc-gen-openapi didn't work for me as well. I tried using gnostic one, but was unsuccessful, there are some tutorials as well, but they mention OpenAPI to gRPC.